gcip/core/job.py

```diff
@@ -200,16 +200,16 @@ def _extend_name(self, name: Optional[str]) -> None:
         if name:
             self._name += "-" + name.replace("_", "-")
 
-    def _extend_stage(self, stage: Optional[str]) -> None:
+    def _extend_stage_value(self, stage: Optional[str]) -> None:
         """This method is used by `gcip.core.sequence.Sequence`s to populate the jobs stage."""
         if stage:
             self._stage += "_" + stage.replace("-", "_")
 
-    def _extend_namespace(self, namespace: Optional[str]) -> None:
+    def _extend_stage(self, stage: Optional[str]) -> None:
```

Comment:
Is stage really optional? Maybe stage is mandatory, cause if `_extend_stage` gets called you intentionally want to extend the stage.

Reply:
Yes, the value must be optional, because the method is blindly called by `gcip.core.sequence.Sequence.populated_jobs`, whereas the Sequence does not care if it passes actual values or `None` those `_extend_*` methods. This method is also just for internal use, so it shouldn't confuse users.

Reply:
Ok. 
I doesn't matter if it confuses users or developers. We should have a clean public **and** private API. But I understand what is going on.